velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -244,17 +244,36 @@ void WindowFuzzer::go() {
           generateSortingKeysAndOrders("s", argNames, argTypes);
     }
     const auto partitionKeys = generateSortingKeys("p", argNames, argTypes);
-    const auto [frameClause, isRowsFrame] = generateFrameClause();
     const auto input =
         generateInputDataWithRowNumber(argNames, argTypes, signature);
+
+    // Randomly select if ROWS or RANGE frame
+    auto typeOfFrame =
+        boost::random::uniform_int_distribution<uint32_t>(0, 1)(rng_);
+
     // If the function is order-dependent or uses "rows" frame, sort all input
     // rows by row_number additionally.
-    if (requireSortedInput || isRowsFrame) {
+    if (requireSortedInput || typeOfFrame == 0) {
       sortingKeysAndOrders.push_back(
           SortingKeyAndOrder("row_number", "asc", "nulls last"));
       ++stats_.numSortedInputs;
     }
 
+    bool isKRangeFrame = false;
+    // typeOfFrame:
+    // 0 - ROWS frame
+    // 1 - RANGE frame
+    // 2 - K-RANGE frame
+    // (K-Range frames require order by keys to only have size of 1)
+    // If it is a range frame, order by keys is size == 1, then 50% chance to
+    // generate k-range frames
```

Comment:
Not sure why 50% chance is needed here, since the frame bound itself is randomly selected. We can just add k-range frames to the list of possible frame bounds based on the number of sorting keys and frame type.

Reply:
@pramodsatya I added 50% chance here so we can also test for the scenario of unbounded RANGE frames with only 1 order by key

Reply:
Got it, but since the number of sorting keys is generated randomly this case should be covered by the existing code, please CMIW.

Reply:
@pramodsatya 
I don't think the number of sorting keys being randomly generated is enough to cover if I want to test for both the scenarios of unbounded range frame and bounded k-range frame. 

```
// 50% chance that there will only be 1 order by key so k-RANGE frames can be
  // generated
  if (vectorFuzzer_.coinToss(0.5)) {
    numKeys = 1;
  } else {
    numKeys = boost::random::uniform_int_distribution<uint32_t>(2, 5)(rng_);
  }
  ```
 
The else statement here would allow for unbounded range frame or rows frame only. The if statement here would limit the number of sorting keys to be 1. This would only guaranteed and satisfy the requirement of k-range frames needing to have only 1 order by key. But unbounded range frames are also allowed to have size of 1 for order by keys.

This is why I have the `if` code block below in order to allow for either test scenarios of unbounded or bounded k-range frame with order by keys of size 1.

Reply:
Unless you wanted me to pass in 2 arguments to `generateFrameClause(typeOfFrame, sizeOfSortingKeys);` ?
If we do it this way, then I can see what you mean by your first comment

Reply:
Yeah I think I will go with this approach since we are already selecting random types of frame bounds in generateFrameClause() function. @pramodsatya What do you think?

Reply:
Thanks for clarifying, the current code lgtm.

Reply:
Implemented suggestion above by Pramod